File: src/app/models/razorpay_model/razorpay.ts
export class RazorpayModel {
  id: string;
  keyId: string;
  secretkey: string;
  businessId: string;

  constructor(params: {
    id: string;
    keyId: string;
    secretkey: string;
    businessId: string;
  }) {
    this.id = params.id;
    this.keyId = params.keyId;
    this.secretkey = params.secretkey;
    this.businessId = params.businessId;
  }

  static fromMap(map: any): RazorpayModel {
    return new RazorpayModel({
      id: map?.id ?? '',
      keyId: map?.key_id ?? '',
      secretkey: map?.secret_key ?? '',
      businessId: map?.business_id ?? '',
    });
  }

  static empty(): RazorpayModel {
    return new RazorpayModel({
      id: '',
      keyId: '',
      secretkey: '',
      businessId: '',
    });
  }
}

File: src/app/models/razorpay_model/razorpay.dart
class RazorpayModel {
  final String id;
  final String keyId;
  final String secretkey;
  final String businessId;

  RazorpayModel(
      {required this.id,
      required this.keyId,
      required this.secretkey,
      required this.businessId});

  factory RazorpayModel.fromMap(Map<String, dynamic> map) {
    return RazorpayModel(
      id: map['id'] ?? '',
      keyId: map['key_id'] ?? '',
      secretkey: map['secret_key'] ?? '',
      businessId: map['business_id'] ?? '',
    );
  }

  factory RazorpayModel.empty() {
    return RazorpayModel(
      id: '',
      keyId: '',
      secretkey: '',
      businessId: '',
    );
  }
}
